uptime_service_validation/coordinator/coordinator.py

```diff
@@ -18,294 +18,292 @@
-                    connection.rollback()
-                    logging.error(ERROR.format(error))
-                finally:
-                    connection.commit()
-                process_loop_count += 1
-                logging.info("Processed it loop count : {0}".format(process_loop_count))
-
-            else:
-                logging.info("Finished processing data from table.")
-            try:
-                updateScoreboard(
-                    connection,
-                    logging,
-                    cur_batch_end,
-                    int(os.environ["UPTIME_DAYS_FOR_SCORE"]),
-                )
+                    createPointRecord(connection, logging, point_record_df)
             except Exception as error:
                 connection.rollback()
                 logging.error(ERROR.format(error))
             finally:
                 connection.commit()
+            process_loop_count += 1
+            logging.info("Processed it loop count : {0}".format(
+                process_loop_count))
 
-            prev_batch_end = cur_batch_end
-            cur_batch_end = prev_batch_end + timedelta(minutes=interval)
-            if prev_batch_end >= cur_timestamp:
-                logging.warning(
-                    "It seems that batch processing took a bit too long than expected as prev_batch_end: %s >= cur_timestamp: %s... progressing to the next batch anyway...",
-                    prev_batch_end,
-                    cur_timestamp,
-                )
+        else:
+            logging.info("Finished processing data from table.")
+        try:
+            updateScoreboard(
+                connection,
+                logging,
+                cur_batch_end,
+                int(os.environ["UPTIME_DAYS_FOR_SCORE"]),
+            )
+        except Exception as error:
+            connection.rollback()
+            logging.error(ERROR.format(error))
+        finally:
+            connection.commit()
+        return bot_log_id
+
+
+def main():
+    process_loop_count = 0
+    load_dotenv()
+
+    logging.basicConfig(
+        stream=sys.stdout,
+        level=logging.INFO,
+        format="%(asctime)s - %(name)s - %(levelname)s - %(message)s",
+    )
+
+    connection = psycopg2.connect(
+        host=os.environ["POSTGRES_HOST"],
+        port=os.environ["POSTGRES_PORT"],
+        database=os.environ["POSTGRES_DB"],
+        user=os.environ["POSTGRES_USER"],
+        password=os.environ["POSTGRES_PASSWORD"],
+    )
+
+    # Step 1 Get previous record and build relations list
+    interval = int(os.environ["SURVEY_INTERVAL_MINUTES"])
+    prev_batch_end, cur_batch_end, bot_log_id = getBatchTimings(
+        connection, logging, interval
+    )
+    while True:
+        bot_log_id = process(connection, prev_batch_end, cur_batch_end, bot_log_id)
+        prev_batch_end = cur_batch_end
+        cur_batch_end = prev_batch_end + timedelta(minutes=interval)
+        if prev_batch_end >= cur_timestamp:
```

Comment:
My vscode complaints about this not being defined... I guess it is now initialized in the `process` method.

Reply:
Fixed in 3a0e2664dbc230a356ca0c311cab34617dd6ab36/